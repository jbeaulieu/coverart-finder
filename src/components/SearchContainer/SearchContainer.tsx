import { Button } from '@mui/material';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import { useState } from 'react';

export type Props = {
  doSearch: (searchParam: string) => void;
};

const SearchContainer = (props: Props) => {

  const { doSearch } = props;
  const [searchParam, setSearchParam] = useState('');

  const handleSubmit = () => doSearch(searchParam);

  return (
    <Box
      component="form"
      sx={{ '& > :not(style)': { m: 1, width: '25ch' } }}
      noValidate
      autoComplete="off"
      onSubmit={(event) => event.preventDefault()}
    >
      <TextField
        variant="outlined"
        id="search-container"
        label="Search by Artist / Album"
        value={searchParam}
        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
          setSearchParam(event.target.value);
        }}
      />
      <Button type="submit" variant="contained" onClick={handleSubmit}>Go</Button>
    </Box>
  );
}

export default SearchContainer;
