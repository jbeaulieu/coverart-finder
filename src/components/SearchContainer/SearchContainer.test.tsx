import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import SearchContainer from './SearchContainer';
import type { Props } from './SearchContainer';

const doSearch = vi.fn();

const defaultProps: Props = {
  doSearch
};

const renderSearchContainer = (props = defaultProps) => {
  return render(<SearchContainer {...props} />);
};

describe('Component SearchContainer', () => {

  it('renders without crashing', () => {
    const { asFragment } = renderSearchContainer();

    expect(asFragment()).toMatchSnapshot();
  });

  describe('on submit', () => {
    it('invokes search function', () => {
      renderSearchContainer();

      const searchField = screen.getByLabelText("Search by Artist / Album");
      const button = screen.getByRole("button", { name: "Go"});

      fireEvent.change(searchField, { target: { value: 'Meat Loaf' } });
      fireEvent.click(button);

      expect(doSearch).toHaveBeenCalledExactlyOnceWith('Meat Loaf');
    });
  });
});
