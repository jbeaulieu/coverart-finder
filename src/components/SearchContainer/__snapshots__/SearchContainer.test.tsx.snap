// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component SearchContainer > renders without crashing 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="MuiBox-root css-xn6ruc"
    novalidate=""
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="search-container"
        id="search-container-label"
      >
        Search by Artist / Album
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="search-container"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1n64csd-MuiNotchedOutlined-root"
          >
            <span>
              Search by Artist / Album
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Go
    </button>
  </form>
</DocumentFragment>
`;
